message_ix/model/MESSAGE/model_core.gms

```diff
@@ -572,6 +584,47 @@ $macro COMMODITY_BALANCE(node,commodity,level,year,time) (
 * export from node and input into technologies located at 'location' taking from 'node' and 'time2' taking from 'time'
         - input(location,tec,vintage,year,mode,node,commodity,level,time2,time)                                        \
         * duration_time_rel(time,time2) * ACT(location,tec,vintage,year,mode,time2) )                                  \
+* +++++
+* commodity input and output associated with construction of new technology capacity (during vintage period)
+  + SUM( (location,tec)$( inv_tec(tec) AND map_tec(location,tec,year) ),                                               \
+* output by all new capacity of technologies located at 'location' sending to 'node' and 'time'
+        output_cap_new(location,tec,year,node,commodity,level,time)                                                    \
+        * CAP_NEW(location,tec,year)                                                                                   \
+* input by all new capacity of technologies located at 'location' taking from 'node' and 'time'
+        - input_cap_new(location,tec,year,node,commodity,level,time)                                                   \
+        * CAP_NEW(location,tec,year) )                                                                                 \
+* commodity input and output associated with retirement of technology capacity (via differentials of capacity of successive periods)
+* for first model period (differential with historical remaining capacity)
+  + SUM( (location,tec,vintage,year2)$( inv_tec(tec) AND map_tec_lifetime(location,tec,vintage,year2)                  \
+            AND first_period(year) AND seq_period(year2,year) ),                                                       \
+* output by all new capacity of technologies located at 'location' sending to 'node' and 'time' distributed over years of periods
+        output_cap_ret(location,tec,vintage,node,commodity,level,time)                                                 \
+        * ( remaining_capacity(node,tec,vintage,year2) - CAP(location,tec,vintage,year) )                              \
```

Comment:
The remaining capacity seems to be a binary value, does it make sense to perform a subtraction such as 
( remaining_capacity(node,tec,vintage,year2) - CAP(location,tec,vintage,year) )?

Reply:
Here we can use something like: 

* output by all new capacity of technologies located at 'location' sending to 'node' and 'time' distributed over years of periods
```
        output_cap_ret(location,tec,vintage,node,commodity,level,time)                                                 
        ( historical_new_capacity(node,tec,vintage) - CAP(location,tec,vintage,year)/ duration_period(year) )  
```      
                                                                                         
* input by all new capacity of technologies located at 'location' taking from 'node' and 'time' distributed over years of periods
```
        - input_cap_ret(location,tec,vintage,node,commodity,level,time)                                               
        (historical_new_capacity(node,tec,vintage) - CAP(location,tec,vintage,year)/ duration_period(year))) 
```